tests/optimize/test_backtest_detail.py

```diff
@@ -457,6 +457,50 @@
     trades=[BTrade(sell_reason=SellType.TRAILING_STOP_LOSS, open_tick=1, close_tick=3)]
 )
 
+# Test 29: trailing_stop should be triggered by low of next candle, without adjusting stoploss using
+# high of stoploss candle.
+# stop-loss: 10%, ROI: 10% (should not apply)
+tc29 = BTContainer(data=[
+    # D   O     H     L     C    V    B  S
+    [0, 5000, 5050, 4950, 5000, 6172, 1, 0],
+    [1, 5000, 5050, 5000, 4900, 6172, 0, 0],    # enter trade (signal on last candle)
+    [2, 4900, 5250, 4500, 5100, 6172, 0, 0],    # Triggers trailing-stoploss
```

Comment:
I'm ... not sure this test does what the description says.

candle 2 (this line) raises the stop - and immediately triggers it (you can easily test this by changing High of this candle).

Based on the description, i'd expect this candle to adjust (but not trigger - so low should be closer to high) - and the consecutive candle should then trigger stop - without adjusting (which implies high should be higher than this high).

Reply:
Maybe i worded description wrong. It looks to me test does what i intended it to do. Here is what happens:

Candle 1: enter trade, adjust stoploss using candle high, stoploss is at 4898.5 price point
Candle 2: stoploss is above candle low (4898.5 > 4500), do not adjust trailing stop, trigger stoploss at 4898.5

Reply:
I debugged and realized now what i saw.
i've changed "high" to significantly higher (5550 instead of 5250) - which caused ROI to trigger.

Basically - i ran into the following Backtesting assumption:

> ROI applies before trailing-stop, ensuring profits are "top-capped" at ROI if both ROI and trailing stop applies

Which has very little to do with this change - except for me not realizing the impact of my value :laughing: